Apps/JobScheduler/src/Jobs/CommunicationJob.cs

```diff
@@ -0,0 +1,167 @@
+
+            IConfigurationSection emailJobSection = this.configuration.GetSection("EmailJob");
+            this.retryFetchSize = emailJobSection.GetValue<int>("MaxRetryFetchSize", 250);
+            IConfigurationSection commEmailJobSection = this.configuration.GetSection("CreateCommEmailsForNewCommunications");
+            this.fromEmailAddressHGDonotreply = commEmailJobSection.GetValue<string>("FromEmailAddressHGDonotreply");
+        }
+
+        /// <inheritdoc />
+        [DisableConcurrentExecution(ConcurrencyTimeout)]
+        public void CreateCommunicationEmailsForNewCommunications()
+        {
+            this.logger.LogDebug($"Creating emails & communication emails...");
+            List<Communication> communications = this.communicationDelegate.GetEmailCommunicationsInNewProcessingOrError();
+
+            if (communications.Count > 0)
+            {
+                this.logger.LogInformation($"Found {communications.Count} communications which are in New status.");
+
+                foreach (Communication comm in communications)
+                {
+#pragma warning disable CA1031 //We want to catch exception.
+                    try
+                    {
+                        DateTime? createdOnOrAfterFilter = null;
+                        string lastProcessedProfileHdid = string.Empty;
+                        List<UserProfile>? usersToSendCommEmails = null;
+                        do
+                        {
+                            if (usersToSendCommEmails != null && usersToSendCommEmails.Count > 0)
+                            {
+                                lastProcessedProfileHdid = usersToSendCommEmails[usersToSendCommEmails.Count - 1].HdId;
+                            }
+
+                            usersToSendCommEmails = this.commEmailDelegate.GetActiveUserProfilesWithoutCommEmailByCommunicationIdAndByCreatedOnOrAfter(comm.Id, createdOnOrAfterFilter, this.retryFetchSize);
+                            foreach (UserProfile profile in usersToSendCommEmails)
+                            {
+                                // Insert a new Email record into db.
+                                Email email = new Email()
+                                {
+                                    From = this.fromEmailAddressHGDonotreply,
+                                    To = profile.Email,
+                                    Subject = comm.Subject,
+                                    Body = comm.Text,
+                                    FormatCode = EmailFormat.HTML,
+                                    Priority = comm.Priority,
+                                };
+                                this.emailQueueService.QueueNewEmail(email, false);
+
+                                // Inser a new CommunicationEmail record into
+                                CommunicationEmail commEmail = new CommunicationEmail()
+                                {
+                                    Email = email,
```

Comment:
This email object does not have an ID. Does that work when adding it to the table on line 118?

Reply:
Yes that works, the DB creates the ID and EF gets the ID when it inserts it and then uses it for inserts down the line.

Reply:
Yes, I have tested this as well, It works fine. As @sslaws mentioned, EF core knows to track these objects.